### PR TITLE
Getting input data from terminal

### PR DESCRIPTION
**Pull request by anastasiakoroto**
### Changes

* The opportunity to input data from terminal is added. User can input his data after the script name. It can be path to file or link, which have an information in json-format, or string in single quote in json-format, that looks like this:
```
-f [path_to_file]
-u [url]
-j [string_in_json_format]
```
* Functions are divided into other functions, that helps to avoid code repeat and does the code more readable.
* Exceptions corrected to get round problems with input data.
* The program is divided into different files.

### CODE REVIEW

File: JSON_serializer/main.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+                        help='decode single quote string in json format from json to object')
+    parser.add_argument('unknown_data', nargs='*', type=str, help='call instruction')
+    return parser
+
+
+def is_valid_file(arg):
+    if not os.path.exists(arg):
+        print(f'The file {arg} does not exist. Check your input data.')
+        return False
+    else:
```
Comment (melnichevv) [inline JSON_serializer/main.py]: this `else:` is not needed here, you can simply `return True` here instead

File: JSON_serializer/main.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        return True
+
+
+def is_url(address):
+    try:
+        url = requests.head(address)
+        return url.status_code == 200
+    except IOError:
+        print(f'The url {address} is not correct. Check your input data. ')
```
Comment (melnichevv) [inline JSON_serializer/main.py]: This error message is too implicit. User will see this message when he/she has problems with connection, for example. Lack of internet connection doesn't mean that URL is incorrect, isn't it?

File: JSON_serializer/main.py [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+        return False
+
+
+def file_argument(file_name):
+    file = open(file_name, 'r', encoding='utf-8')
```
Comment (melnichevv) [inline JSON_serializer/main.py]: `file` is a builtin variable in Python. That's not a best practice to redefine it. It's better to use another name for this variable: eg. simple `f`.
Also you can use context manager for opening files

File: JSON_serializer/main.py [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+        temp = line.strip()
+        string_from_file += temp
+    if string_from_file != '':
+        obj = serializer.check_format(string_from_file)
+        write_to_file(obj)
+        file.close()
+
+
+def write_to_file(decoded_string):
+    if len(str(decoded_string)) < 500:
```
Comment (melnichevv) [inline JSON_serializer/main.py]: what is 500? do you think it worth adding this to some kind of a settings file?

File: JSON_serializer/main.py [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+        print(decoded_string)
+    else:
+        out_file = open('output.txt', 'w', encoding='utf-8')
+        out_file.write(str(decoded_string))
+        print('Deserialized object is too long. It was written to output.txt in this directory.')
+        answer = input('Do you want to open this file in terminal? (y/n)')
+        if answer in ['y', 'yes', 'Y', 'YES']:
```
Comment (melnichevv) [inline JSON_serializer/main.py]: `['y', 'yes', 'Y', 'YES']` this is constant block, isn't it? May be it would be good to move this constants file?

**Review by melnichevv — CHANGES_REQUESTED** [review]
Overall looks good! 
_Haven't played with the script again. Will do it a bit later today. You can work on these changes in a meantime_

File: JSON_serializer/const.py [hunk 6]
```diff
@@ -0,0 +1,2 @@
+ANSWERS = ['y', 'Y', 'YES', 'yes', 'Yes']
```
Comment (melnichevv) [inline JSON_serializer/const.py]: this is not just answers, it's like positive answers, so `POSITIVE_ANSWERS` sounds better, isn't it?

Comment (anastasiakoroto) [reply JSON_serializer/const.py]: Agree, i'll correct it.

**Review by melnichevv — APPROVED** [review]

**Pull request merged**